torch/csrc/autograd/python_function.cpp

```diff
@@ -310,38 +301,31 @@ using t2var_type = std::unordered_map<PyObject *, THPVariable *>;
 
 // Bump the counters of all recorded dirty input tensors, adding each of them
 // into dirty_inputs.  Also does some sanity checking.
-static void _mark_dirty(THPFunction *self, t2var_type &t2var,
-        std::unordered_set<PyObject *> &dirty_inputs)
+static std::vector<PyObject*> _parse_dirty_inputs(THPFunction *self)
```

Comment:
nit: I think `_mark_dirty` would still be a better name, because it implies that it has a side effect, while `parse_dirty_inputs` would imply that it doesn't.